addons/binding/org.openhab.binding.globalcache/src/main/java/org/openhab/binding/globalcache/handler/GlobalCacheHandler.java

```diff
@@ -66,14 +71,17 @@
 
     private InetAddress ifAddress;
     private CommandProcessor commandProcessor;
-    ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(MAX_GC_DEVICES);
-    ScheduledFuture<?> scheduledFuture;
+    private ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(MAX_GC_THREADS);
```

Comment:
If you need that many threads for every single handler (sigh...), I guess it makes most sense to indeed create a separately named pool for each, because otherwise you will run out of available threads quickly if you have more than a single handler.